erts/emulator/beam/external.c

```diff
@@ -3023,7 +3054,51 @@ dec_pid(ErtsDistExternal *edep, ErtsHeapFactory* factory, const byte* ep,
 #define ENC_BIN_COPY ((Eterm) 3)
 #define ENC_MAP_PAIR ((Eterm) 4)
 #define ENC_HASHMAP_NODE ((Eterm) 5)
-#define ENC_LAST_ARRAY_ELEMENT ((Eterm) 6)
+#define ENC_STORE_MAP_ELEMENT ((Eterm) 6)
+#define ENC_START_SORTING_MAP ((Eterm) 7)
+#define ENC_CONTINUE_SORTING_MAP ((Eterm) 8)
+#define ENC_PUSH_SORTED_MAP ((Eterm) 9)
+#define ENC_LAST_ARRAY_ELEMENT ((Eterm) 10)
+
+static Eterm* alloc_map_array(Uint size)
+{
+    return (Eterm *) erts_alloc(ERTS_ALC_T_T2B_DETERMINISTIC,
+                                size * 2 * sizeof(Eterm));
+}
+
+static int map_key_compare(Eterm *a, Eterm *b)
+{
+    Sint c = CMP_TERM(*a, *b);
+    if (c < 0) {
+        return -1;
+    } else if (c > 0) {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+#ifdef INCLUDE_YCF_TRANSFORMED_ONLY_FUNCTIONS
+static void yielding_qsort(Eterm *map_array, Uint n)
+{
+    erts_qsort(map_array, n/2, 2*sizeof(Eterm),
+               (int (*)(const void *, const void *)) map_key_compare);
+}
```

Comment:
One could call `erts_qsort_ycf_gen_yielding` dirctly so this wrapper is not needed. `erts_qsort_ycf_gen_yielding` has a prototype in  [global.h](https://github.com/erlang/otp/blob/master/erts/emulator/beam/global.h#L1408). Having the wrapper is of course a matter of taste and I don't see any reason not to have it other than that YCF does not need to do any work for this file if the rapper does not exist.